OPAL/br/src/main/scala/org/opalj/br/fpcf/properties/fieldaccess/FieldAccessInformation.scala

```diff
@@ -0,0 +1,210 @@
+    with FieldAccessInformationPropertyMetaInformation[S] {
+
+    protected val encodedDirectAccesses: LongLinkedSet
+    protected val encodedIndirectAccessReceivers: LongMap[AccessReceiver] // Caller Context and PC => Receiver
+    protected val encodedIndirectAccessParameters: LongMap[AccessParameter] // Caller Context and PC => Parameter
+
+    def directAccesses(implicit declaredMethods: DeclaredMethods): Iterator[(DefinedMethod, PC)] =
+        encodedDirectAccesses.iterator.map(decodeDirectFieldAccess(_))
+
+    def indirectAccesses: Iterator[(Int, PC, AccessReceiver, AccessParameter)] = {
+        encodedIndirectAccessReceivers.iterator.map {
+            kv =>
+                {
+                    val (fa: FieldAccess, receiver: Option[(ValueInformation, br.PCs)]) = kv
+                    val parameter = encodedIndirectAccessParameters.getOrElse(fa, None)
+                    val decodedFieldAccess = decodeIndirectFieldAccess(fa)
+                    (decodedFieldAccess._1, decodedFieldAccess._2, receiver, parameter)
+                }
+        }
+    }
+
+    def getNewestNDirectAccesses(n: Int)(implicit declaredMethods: DeclaredMethods): Iterator[(DefinedMethod, PC)] =
+        encodedDirectAccesses.iterator.take(n).map(decodeDirectFieldAccess(_))
+
+    def getNewestNIndirectAccesses(n: Int): Iterator[(Int, PC, AccessReceiver, AccessParameter)] = {
+        encodedIndirectAccessReceivers.iterator.take(n).map {
+            kv =>
+                {
+                    val (fa: FieldAccess, receiver: Option[(ValueInformation, br.PCs)]) = kv
+                    val parameter = encodedIndirectAccessParameters.getOrElse(fa, None)
+                    val decodedFieldAccess = decodeIndirectFieldAccess(fa)
+                    (decodedFieldAccess._1, decodedFieldAccess._2, receiver, parameter)
+                }
+        }
+    }
+
+    def numAccesses: Int = numDirectAccesses + numIndirectAccesses
+
+    def numDirectAccesses: Int = encodedDirectAccesses.size
+
+    def numIndirectAccesses: Int = encodedIndirectAccessReceivers.valuesIterator.map { _.size }.sum
+
+    def included(other: S): S = included(other, 0)
+
+    def included(other: S, seenAccesses: Int): S
+
+    protected def included(
+        other:           S,
+        seenAccesses:    Int,
+        propertyFactory: (LongLinkedSet, LongMap[AccessReceiver], LongMap[AccessParameter]) => S
+    ): S = {
+        var newAccesses = encodedDirectAccesses
+        other.encodedDirectAccesses.forFirstN(other.numDirectAccesses - seenAccesses)(newAccesses += _)
+        val newReceivers = encodedIndirectAccessReceivers ++ other.encodedIndirectAccessReceivers
+        val newParameters = encodedIndirectAccessParameters ++ other.encodedIndirectAccessParameters
+
+        if (newAccesses == encodedDirectAccesses && newReceivers == encodedIndirectAccessReceivers && newParameters == encodedIndirectAccessParameters)
```

Comment:
These equality checks might be expensive. Did you check whether they could be replaced with identity checks (i.e., do the collections return an identity if nothing was added)?

Reply:
The collection used for direct accesses does, i have updated the check accordingly. The LongMaps used for indirect accesses do not specify this, and reading their implementation i am quite not sure, so i would leave them as equality checks for the time being.